File: Interpreted/ackermann-function/ack.pl
sub A {
  my ($m, $n) = @_;
  $m == 0 ? $n + 1 :
  $n == 0 ? A($m - 1, 1) :
            A($m - 1, A($m, $n - 1))
}

print A($ARGV[0], $ARGV[1]), "\n";

File: Interpreted/ackermann-function/ack.exs
defmodule Ackermann do
  def ack(0, n), do: n + 1
  def ack(m, 0), do: ack(m - 1, 1)
  def ack(m, n), do: ack(m - 1, ack(m, n - 1))
end

[n | tail] = System.argv
[m | _] = tail

{n, _} = Integer.parse(n)
{m, _} = Integer.parse(m)

IO.puts Ackermann.ack(n, m)
